File: backend/src/main/java/hu/szakdoga/backend/timetable/data/dto/SubTaskDTO.java
package hu.szakdoga.backend.timetable.data.dto;

public class SubTaskDTO {
    public final Long id;
    public final String name;
    public final boolean fulfilled;
    public final Long mainTaskId;

    public SubTaskDTO(Long id, String name, boolean fulfilled, Long mainTaskId) {
        this.id = id;
        this.name = name;
        this.fulfilled = fulfilled;
        this.mainTaskId = mainTaskId;
    }
}

File: frontend/src/app/shared/model/timetable/dto/sub-task.dto.ts
export class SubTaskDto {
    public readonly id: number | null;
    public readonly name: string;
    public readonly fulfilled: boolean;
    public readonly mainTaskId: number;

    constructor(name: string, fulfilled: boolean, mainTaskId: number, id: number | null = null) {
        this.id = id;
        this.name = name;
        this.fulfilled = fulfilled;
        this.mainTaskId = mainTaskId;
    }
}
